### PR TITLE
feat(discover): Support arbitrary aggregate arguments

### PR DESCRIPTION
**Pull request by wmak**
- Currently the `AGGREGATE_PATTERN` only supports up to two arguments,
  with the second argument becoming a string with commas included
  - eg. `foo(a,b,c,d)` is parsed as:
   - function: foo
   - first arg: a
   - second arg: b,c,d
- Copying parse_arguments from the backend to the frontend so that we
  can be consistent with how this works between both sides

### CODE REVIEW

File: static/app/utils/discover/fields.tsx [hunk 1]
```diff
@@ -616,16 +621,88 @@ export function getMeasurementSlug(field: string): string | null {
...
+      // begun an escape character, we should end it
+      escaped = false;
+    } else if (quoted && columnText[j] === ',') {
+      // when we are inside a quoted string and see
+      // a comma, it should not be considered an
+      // argument separator
+    } else if (columnText[j] === ',') {
+      // when we see a comma outside of a quoted string
+      // it is an argument separator
+      args.push(columnText.substr(i, j - i).trim());
```
Comment (Zylphrex) [inline static/app/utils/discover/fields.tsx:692]: Nit: A bit less roundabout and is closer to what the backend does

```suggestion
      args.push(columnText.substring(i, j).trim());
```

File: static/app/utils/discover/fields.tsx [hunk 2]
```diff
@@ -616,16 +621,88 @@ export function getMeasurementSlug(field: string): string | null {
...
+      i = j + 1;
+    }
+    j += 1;
+  }
+
+  if (i !== j) {
+    // add in the last argument if any
+    args.push(columnText.substr(i).trim());
```
Comment (Zylphrex) [inline static/app/utils/discover/fields.tsx:700]: Nit: Just to be consistent with the use of `substr` vs `substring`

```suggestion
    args.push(columnText.substring(i).trim());
```

File: static/app/utils/discover/fields.tsx (hunk 1 above)
Comment (wmak) [reply static/app/utils/discover/fields.tsx:692]: ... dunno why I used substr its deprecated even, will update :+1:

**Review by Zylphrex — APPROVED** [review]
I hope we can eventually address this duplication between the frontend and the backend. But otherwise LGTM

**Pull request merged**